jobs/src/main/scala/dcos/metronome/utils/state/MarathonStore.scala

```diff
@@ -0,0 +1,86 @@
+package dcos.metronome.utils.state
+
+import com.google.protobuf.InvalidProtocolBufferException
+import mesosphere.marathon.StoreCommandFailedException
+import org.slf4j.LoggerFactory
+
+import scala.collection.immutable.Seq
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+import scala.util.control.NonFatal
+
+class MarathonStore[S <: MarathonState[_, S]](
+  store:    PersistentStore,
+  newState: () => S,
+  prefix:   String)(implicit ct: ClassTag[S]) extends EntityStore[S] {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+  private[this] val log = LoggerFactory.getLogger(getClass)
+
+  private[this] lazy val lockManager = LockManager.create()
+
+  def fetch(key: String): Future[Option[S]] = {
+    log.debug(s"Fetch $prefix$key")
+    store.load(prefix + key)
+      .map {
+        _.map { entity =>
+          stateFromBytes(entity.bytes.toArray)
+        }
+      }
+      .recover {
+        case ipe: InvalidProtocolBufferException =>
+          log.warn(s"Unable to read $key due to a protocol buffer exception")
+          None
+        case NonFatal(ex) =>
+          throw new StoreCommandFailedException(s"Could not fetch ${ct.runtimeClass.getSimpleName} with key: $key", ex)
+      }
+  }
+
+  def modify(key: String, onSuccess: (S) => Unit = _ => ())(f: Update): Future[S] = {
+    lockManager.executeSequentially(key) {
```

Comment:
Marathon uses `mesosphere.marathon.KeyedLock` to sequentialize writes and reads. Checkout the [BasePersistenceStore](https://github.com/mesosphere/marathon/blob/releases/1.5/src/main/scala/mesosphere/marathon/core/storage/store/impl/BasePersistenceStore.scala#L44).

Reply:
we will be migrating out of this persistence anyway - probably...

Reply:
+1 what @alenkacz says :)